feat(component): created reusable button

ngrujicic opened the pull request.

ngrujicic committed:
feat(component): created reusable button

markoMVP85 commented on src/components/atom/button/index.tsx:
```diff
@@ -0,0 +1,22 @@
+const Button = ({ text, hasAddIcon, onClick, type, disabled, styles, ...rest }: ButtonProps) => {
+  return (
+    <button className={styles} onClick={onClick} disabled={disabled} type={type} {...rest}>
+      {hasAddIcon && <AddIcon />}
```
We should create the abstract component. Right now u have created a hardcoded icon but u have to allow adding other icons depending on features. Eg: Somewhere u will need the "-" icon for removing or smth else. When we create a component we should make it reusable for all cases.

markoMVP85 replied:
Also, try to avoid {hasAddIcon && <AddIcon />}. Rather use {hasAddIcon ? <AddIcon /> : null} This is one of the most common mistakes. React does not render "false" which will be the first case. This can lead to a small performance overhead if the component tree is complex and deeply nested.

markoMVP85 commented on src/components/atom/button/index.tsx:
```diff
@@ -0,0 +1,22 @@
+
+const Button = ({ text, hasAddIcon, onClick, type, disabled, styles, ...rest }: ButtonProps) => {
+  return (
+    <button className={styles} onClick={onClick} disabled={disabled} type={type} {...rest}>
```
One suggestion for future cases. When u have to disable the component u have to prevent click or some other actions. Right now if I go to the inspect element in the browser and remove disable property, I can easily click on btn and trigger the action. In order to prevent that u have to make smth like this 
onClick={() => !disabled ? onClick() : null} With this u will prevent the prev case cuz on the mount component we will check if we are gonna append some action to the button or we are not. Even though u removed disabled from the inspect element, the component is already mounted without the "click" action

npanajotov replied:
Please use className instead of styles, because this component receives classes not styles.

markoMVP85 commented on src/components/atom/button/index.tsx:
```diff
@@ -0,0 +1,22 @@
+  styles: string;
+};
+
+const Button = ({ text, hasAddIcon, onClick, type, disabled, styles, ...rest }: ButtonProps) => {
```
type should be default button

npanajotov replied:
Default type should be a button - to prevent submit behavior.

markoMVP85 commented on src/pages/HomePage/index.tsx:
```diff
@@ -1,4 +1,17 @@
+        type="button"
+        text="{Enter button text}"
+        onClick={() => console.log("click")}
+        styles="button-primary-small"
```
this is not a style it is className. My suggestion is to have a default style and with additional tw classes, u can change it. So there is no need for the extra lines in the css

markoMVP85 commented on src/pages/HomePage/index.tsx:
```diff
@@ -1,4 +1,17 @@
+        onClick={() => console.log("click")}
+        styles="button-primary-small"
+        hasAddIcon={true}
+        disabled={false}
```
U do not need to add  disabled={false}. It should be disabled = false by default and if u want to disable it just add disable property without  disabled={true} just disabled

markoMVP85 replied:
The icon should be ReactComponent not boolean. I explained in the prev comment

npanajotov commented on src/components/atom/button/styles.scss:
```diff
@@ -0,0 +1,65 @@
+  cursor: pointer;
+
+  &-primary {
+    @extend .button;
```
If you are using scss, you don't need to use @extend for getting parent style.

npanajotov commented on src/components/atom/button/styles.scss:
```diff
@@ -0,0 +1,65 @@
+    background-color: #e6007a !important;
+
+    &:hover {
+      background-color: black !important;
```
Avoid to use !important.

npanajotov commented on src/components/atom/button/styles.scss:
```diff
@@ -0,0 +1,65 @@
+
+  &-interactive {
+    @extend .button-primary;
+    width: 424px;
```
Please use proportional width/height instead of hard coded values. In this case, you can put padding for height.

ngrujicic committed:
feat(component): changed style to tailwind

ngrujicic committed:
feat(component): added select from modal button

ngrujicic committed:
style(component): removed text-base

npanajotov commented on src/components/atom/button/index.tsx:
```diff
@@ -0,0 +1,112 @@
+
+type ButtonProps = {
+  text?: string;
+  onClick: () => void;
```
onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;

npanajotov commented on src/components/atom/button/index.tsx:
```diff
@@ -0,0 +1,112 @@
+      case "primary":
+        return (
+          <button
+            className={`${getButtonSizeStyle()} flex items-center justify-center 
```
Use classNames instead custom function.
https://www.npmjs.com/package/classnames

npanajotov commented on src/components/atom/button/index.tsx:
```diff
@@ -0,0 +1,112 @@
+          <button
+            className={`flex h-[31px] w-[145px] items-center justify-center gap-2
+              rounded-[100px] bg-[#e6007a] px-3 py-[6px] text-[16px] font-normal text-white
+              hover:bg-black ${className ? className : null}`}
```
{className || ""}

ngrujicic committed:
refactor(component): refactored button component to use classnames

exported tailwind classes to external scss file and added classnames to button component to combine
the styles properly

ngrujicic committed:
style(component): removed fixed width and height

ngrujicic committed:
style(component): added styles from tailwind config

npanajotov committed:
fix: fixed button styles, refactor folder structure, update eslint"
